### PR TITLE
Remove DirectRunner implementations from DoFnTester

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---
Part of the removal of runners-core from the Core SDK, and the deletion
of the DirectRunner from the core SDK.

The implementations of DoFn.Context and DoFn.ProcessContext are
a simplified version of the DoFnRunnerBase versions of the same that
use local data structures.

### CODE REVIEW

**Comment by tgroh** [comment]
R: @kennknowles

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java [hunk 1]
```diff
@@ -239,7 +248,11 @@ public void finishBundle() {
     if (state == State.UNSTARTED) {
       startBundle();
     }
-    fnRunner.finishBundle();
+    try {
+      fn.finishBundle(createContext(fn));
+    } catch (Exception e) {
+      throw UserCodeException.wrap(e);
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java]: `DoFnTester` is not expected to be run via a runner, but for directly unit testing a `DoFn`. So I don't think that it should wrap all these exceptions. Am I wrong?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java [hunk 2]
```diff
@@ -372,11 +383,15 @@ public T apply(WindowedValue<T> input) {
    * Returns the value of the provided {@link Aggregator}.
    */
   public <AggregateT> AggregateT getAggregatorValue(Aggregator<?, AggregateT> agg) {
+    return extractAggregatorValue(agg, agg.getCombineFn());
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java]: Seems appropriate to pull out the `CombineFn` inside `extractAggregatorValue`, yea? Is there any meaning to passing any other value for this second parameter?

Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java]: ExtractAggregatorValue exists only to bind the accumulator type; instead, this should probably be

`return extractAggregatorValue(agg.getName(), agg.getCombineFn())`

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java (hunk 1 above)
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java]: This is approximately equal to
```
try {
  fn.startBundle();
} catch (Exception e) {
  throw new RuntimeException(e);
}
```

as `DoFn#processElement()` is declared to throw an exception. Alternatively, this method could be declared to throw an exception (which is probably reasonable), although it puts a slightly higher burden on test writers.

Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java]: Throwing the exception directly./

**Comment by kennknowles** [comment]
LGTM. Will squash and merge.

**Pull request closed without merging**